django/apps/config.py

```diff
@@ -214,3 +214,8 @@ def ready(self):
         """
         Override this method in subclasses to run code when Django starts.
         """
+
+    @property
+    def model_default_pk(self):
+        from django.conf import settings
+        return settings.MODEL_DEFAULT_PK
```

Comment:
I feel like this setting is out of context here. Should `AppConfig` only provide information in the context of an app?

If I want to find out what kind of primary key a given app is asking for, I could look for `AppConfig.model_default_pk`, but if we're overriding if with a project setting it can cause some confusion.

Reply:
The _default_ model PK field is defined in `settings.MODEL_DEFAULT_PK`, which applies to all applications thanks to this property. This can be overridden on a per-app basis if they ship migrations. Without this if an app ships migrations then changing the `MODEL_DEFAULT_PK` will cause new migrations to be generated, which isn't appropriate for third party apps.